File: src/test/java/no/nav/pensjon/selvbetjeningopptjening/util/net/UriUtilTest.java
package no.nav.pensjon.selvbetjeningopptjening.util.net;

import org.junit.jupiter.api.Test;

import java.net.URI;
import java.net.URISyntaxException;

import static org.junit.jupiter.api.Assertions.*;

class UriUtilTest {

    @Test
    void uriFrom_url_with_port_produces_uri() throws URISyntaxException {
        URI uri = UriUtil.uriFrom("https://www.example.net:80");
        assertEquals("www.example.net", uri.getHost());
        assertEquals(80, uri.getPort());
    }

    @Test
    void uriFrom_url_without_port_causes_URISyntaxException() {
        var exception = assertThrows(URISyntaxException.class, () -> UriUtil.uriFrom("https://www.example.net"));
        assertEquals("No URI port specified: https://www.example.net", exception.getMessage());
    }

    @Test
    void uriFrom_nonUri_causes_URISyntaxException() {
        var exception = assertThrows(URISyntaxException.class, () -> UriUtil.uriFrom("invalid"));
        assertEquals("No URI port specified: invalid", exception.getMessage());
    }
}

File: src/test/java/no/nav/pensjon/selvbetjeningopptjening/tech/web/UriUtilTest.kt
package no.nav.pensjon.selvbetjeningopptjening.tech.web

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class UriUtilTest {

    @Test
    fun `formatAsUri produces a URI given scheme, authority, path`() {
        val uri = UriUtil.formatAsUri("scheme1", "authority1", "path1")
        assertEquals("scheme1://authority1/path1", uri)
    }

    @Test
    fun `formatAsUri produces a URI given scheme, authority, but no path`() {
        val uri = UriUtil.formatAsUri("scheme1", "authority1", "")
        assertEquals("scheme1://authority1", uri)
    }
}
